pkg/querier/tripperware/instantquery/instant_query.go

```diff
@@ -176,7 +183,7 @@ func (instantQueryCodec) EncodeResponse(ctx context.Context, res tripperware.Res
 		return nil, httpgrpc.Errorf(http.StatusInternalServerError, "invalid response format")
 	}
 
-	b, err := json.Marshal(a)
+	contentType, b, err := marshalResponse(a, req.Header.Get("Accept"))
```

Comment:
Should we only try to return non JSON response format if we know the request comes from Ruler?
If the request doesn't come from Ruler I feel we want to enforce JSON response even though they specify protobuf in their protocol like `Accept: protobuf, json` to reduce impact as this protobuf format is Cortex specific.

But other than this way Idk if we have a way to find out Ruler requests.

WDYT? @alanprot @danielblando

Reply:
@yeya24 
One way is to define a new MIME type like `application/x-protobuf-ruler` (just example) and then use it to accept header.

Reply:
Yeah.. maybe we should define our internal MIME type so external calls will only get that format if they really wanna.
I would not add "ruler" in the format though. Maybe something like `application/x-cortex.query.v1+proto` (idk the latest best practices to define those custom Mime types :P )

Reply:
This sounds worth trying it out. Let's hear from other maintainers, too